code/modules/shuttle/super_cruise/orbital_map_components/orbital_objects/lavaland.dm

```diff
@@ -11,4 +11,10 @@
 /datum/orbital_object/z_linked/lavaland/New()
 	. = ..()
 	var/datum/orbital_map/linked_map = SSorbits.orbital_maps[orbital_map_index]
-	linked_map.center = src
+	if(!linked_map.center) // only if there's no centre of the universe
+		linked_map.center = src
+
+/datum/orbital_object/z_linked/lavaland/post_map_setup()
+	// Lavaland planet will orbit the sun when they're not the centre
+	var/datum/orbital_map/linked_map = SSorbits.orbital_maps[orbital_map_index]
+	set_orbitting_around_body(linked_map.center, 3000)
```

Comment:
This won't work since lavaland is designed as a static object due to its gravitational pull

Reply:
Removed now. I will make it work someday...